contracts/cw721-base/src/state.rs

```diff
@@ -1,152 +1,56 @@
-use schemars::JsonSchema;
+use cosmwasm_std::CustomMsg;
+// expose to all others using contract, so others dont need to import cw721
+pub use cw721::state::*;
 use serde::de::DeserializeOwned;
-use serde::{Deserialize, Serialize};
-use std::marker::PhantomData;
-
-use cosmwasm_std::{Addr, BlockInfo, CustomMsg, StdResult, Storage};
-
-use cw721::{ContractInfoResponse, Cw721, Expiration};
-use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
-
-pub struct Cw721Contract<'a, T, C, E, Q>
-where
-    T: Serialize + DeserializeOwned + Clone,
-    Q: CustomMsg,
-    E: CustomMsg,
+use serde::Serialize;
+
+pub struct Cw721Contract<
+    'a,
+    TMetadata,
+    TCustomResponseMessage,
+    TExtensionExecuteMsg,
+    TMetadataResponse,
+    TCollectionInfoExtension,
+> where
+    TMetadata: Serialize + DeserializeOwned + Clone,
+    TMetadataResponse: CustomMsg,
+    TExtensionExecuteMsg: CustomMsg,
+    TCollectionInfoExtension: Serialize + DeserializeOwned + Clone,
 {
-    pub contract_info: Item<'a, ContractInfoResponse>,
-    pub token_count: Item<'a, u64>,
-    /// Stored as (granter, operator) giving operator full control over granter's account
-    pub operators: Map<'a, (&'a Addr, &'a Addr), Expiration>,
-    pub tokens: IndexedMap<'a, &'a str, TokenInfo<T>, TokenIndexes<'a, T>>,
-    pub withdraw_address: Item<'a, String>,
-
-    pub(crate) _custom_response: PhantomData<C>,
-    pub(crate) _custom_query: PhantomData<Q>,
-    pub(crate) _custom_execute: PhantomData<E>,
-}
-
-// This is a signal, the implementations are in other files
-impl<'a, T, C, E, Q> Cw721<T, C> for Cw721Contract<'a, T, C, E, Q>
-where
-    T: Serialize + DeserializeOwned + Clone,
-    C: CustomMsg,
-    E: CustomMsg,
-    Q: CustomMsg,
-{
-}
-
-impl<T, C, E, Q> Default for Cw721Contract<'static, T, C, E, Q>
+    pub config: Cw721Config<
+        'a,
+        TMetadata,
+        TCustomResponseMessage,
+        TExtensionExecuteMsg,
+        TMetadataResponse,
+        TCollectionInfoExtension,
+    >,
+}
```

Comment:
... it is more readable: `Cw721Contract<'a, TMetadata, TCustomResponseMessage, TExtensionExecuteMsg, TMetadataResponse, TCollectionInfoExtension,>`.

Please note, all is stored now in `Cw721Config` (and not in `Cw721Contract` anymore) ...